include/picongpu/algorithms/FieldToParticleInterpolation.hpp

```diff
@@ -53,8 +53,18 @@ struct FieldToParticleInterpolation
 
     PMACC_CASSERT_MSG(
         __FieldToParticleInterpolation_supercell_is_to_small_for_stencil,
-        pmacc::math::CT::min< SuperCellSize >::type::value * GUARD_SIZE >= lowerMargin &&
-        pmacc::math::CT::min< SuperCellSize >::type::value * GUARD_SIZE >= upperMargin
+        pmacc::math::CT::min<
```

Comment:
This type of multi-line construction is now present at several places, e.g. also at EmZ.hpp, several files for Esirkepov. Maybe it is worth to make a compile-time "function" for that and call it with different parameters?

Reply:
I am not sure if this make sense. I am undecided, therefore I will keep it as it is for now. Do you think a helper function with 4 arguments can help?

Reply:
@sbastrakov I think replacing our C++98 metaprogramming constructs in public typdefs/usings (e.g. margins) and asserts would be a wonderful refactoring in general. Documented in: #2623